### PR TITLE
Wizards redesign and cleanup

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Part of the redesign outlined in #704. Updates:

1. Wizard navigation redesign.
2. Removal of secondary navigation – it was used only in Engagement wizard's Commenting tab, but all of it's contents can fit into one screen. A secondary navigation is an overkill.
3. Removal of `Checklist` component and PHP handlers – these were not used. 
4. Multiple refactors and smaller code & design improvements

### How to test the changes in this Pull Request:

1. Visit all the wizards
2. Observe that the functionality is the same, but layout has changed (according to the design in #704)

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by thomasguillot** [comment]
I pushed a few changes https://github.com/Automattic/newspack-plugin/pull/705/commits/cccaf802e197fec76bc79295b159001d73264fc5

I think there are a bunch of stuff that we can get rid of (maybe in a separate PR).

Like the concept of a `card` we don't really need it anymore. Same with the `grid`.

File: assets/components/src/plugin-installer/index.js
Comment (dkoo) [inline assets/components/src/plugin-installer/index.js:177]: This is true on `master` too, but it seems heightened to me with the new design. And maybe now's a good time to change it. 

1. (See the Disqus section for what I'm talking about.) If there's more than one plugin to be installed, should this primary action button say "Install All" instead of just "Install"? With just "install" it's not immediately clear to me how the action differs from clicking "Install" next to one of the individual plugins.

Screen Shot 2020-11-30 at 3 13 34 PM

2. (See the Coral section for what I'm talking about.) If there is is only one plugin to be installed, do we even need the action button, since it will do the same thing as clicking "Install" next to the single plugin?

**Review by dkoo — CHANGES_REQUESTED** [review]
This is looking good, just left a few minor questions which @thomasguillot may want to weigh in on.

File: assets/components/src/tabbed-navigation/style.scss
Comment (thomasguillot) [reply assets/components/src/tabbed-navigation/style.scss:18]: > @thomasguillot is your intention to use the "dropdown" mobile styles you mocked up for the new onboarding screens, that look like this?

It is, yes, but I'm not too worried about it for now.

File: assets/components/src/plugin-installer/index.js
Comment (thomasguillot) [reply assets/components/src/plugin-installer/index.js:177]: I'm not even sure we need the buttons here.

Comment (adekbadek) [reply assets/components/src/plugin-installer/index.js:177]: Removed in 1d4690bd37b54196720977c93e4805951c56b53e

File: assets/components/src/with-wizard-screen/index.js
Comment (adekbadek) [reply assets/components/src/with-wizard-screen/index.js]: Good idea, done in 5705408e7e51ab416e6baf5c8af287f5ccd0e370

**Comment by adekbadek** [comment]
@thomasguillot 

> I pushed a few changes cccaf80

Is the white background on the Dashboard intentional?

image

**Comment by thomasguillot** [comment]
> Is the white background on the Dashboard intentional?

For now yes.

To be honest this will need a complete redesign at some point (and the Dashboard might evolve anyway) so it's fine for now.

**Review by dkoo — APPROVED** [review]
Changes since my last review look good from a technical standpoint. I'll let @thomasguillot approve from a design perspective. Thanks!

Once this lands I'll also continue working on #697 (I had begun but some of the work I started doing was redundant with this branch).

**Comment by adekbadek** [comment]
> I'll let @thomasguillot approve from a design perspective.

I assume https://github.com/Automattic/newspack-plugin/pull/705#event-4056253696 is an approval ;)

image

**Pull request merged**
